Invalid argument to `Span::record` in kube-client

### Discussed in https://github.com/kube-rs/kube/discussions/1068

<div type='discussions-op-text'>

<sup>Originally posted by **jesperpedersen** November  2, 2022</sup>
In https://github.com/pgopr/pgopr I changed 

```
kube = { version = "0.75.0", default-features = true, features = ["client", "derive", "runtime"] }
```

to

```
kube = { version = "0.76.0", default-features = true, features = ["client", "derive", "runtime"] }
```

But I get

```
error[E0308]: mismatched types
    --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/kube-client-0.76.0/src/client/builder.rs:137:57
     |
137  |                         span.record("http.status_code", status.as_u16());
     |                              ------                     ^^^^^^^^^^^^^^^
     |                              |                          |
     |                              |                          expected reference, found `u16`
     |                              |                          help: consider borrowing here: `&status.as_u16()`
     |                              arguments to this function are incorrect
     |
     = note: expected reference `&_`
                     found type `u16`
note: associated function defined here
    --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-0.1.31/src/span.rs:1183:12
     |
1183 |     pub fn record<Q: ?Sized, V>(&self, field: &Q, value: &V) -> &Self
     |            ^^^^^^

error[E0277]: the size for values of type `str` cannot be known at compilation time
    --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/kube-client-0.76.0/src/client/builder.rs:139:34
     |
139  | ...                   span.record("otel.status_code", "ERROR");
     |                            ^^^^^^ doesn't have a size known at compile-time
     |
     = help: the trait `Sized` is not implemented for `str`
note: required by a bound in `Span::record`
    --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-0.1.31/src/span.rs:1183:30
     |
1183 |     pub fn record<Q: ?Sized, V>(&self, field: &Q, value: &V) -> &Self
     |                              ^ required by this bound in `Span::record`

error[E0277]: the size for values of type `str` cannot be known at compilation time
    --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/kube-client-0.76.0/src/client/builder.rs:153:30
     |
153  |                         span.record("otel.status_code", "ERROR");
     |                              ^^^^^^ doesn't have a size known at compile-time
     |
     = help: the trait `Sized` is not implemented for `str`
note: required by a bound in `Span::record`
    --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-0.1.31/src/span.rs:1183:30
     |
1183 |     pub fn record<Q: ?Sized, V>(&self, field: &Q, value: &V) -> &Self
     |                              ^ required by this bound in `Span::record`

error[E0308]: mismatched types
    --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/kube-client-0.76.0/src/client/builder.rs:156:65
     |
156  | ...                   span.record("http.status_code", status.as_u16());
     |                            ------                     ^^^^^^^^^^^^^^^
     |                            |                          |
     |                            |                          expected reference, found `u16`
     |                            |                          help: consider borrowing here: `&status.as_u16()`
     |                            arguments to this function are incorrect
     |
     = note: expected reference `&_`
                     found type `u16`
note: associated function defined here
    --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-0.1.31/src/span.rs:1183:12
     |
1183 |     pub fn record<Q: ?Sized, V>(&self, field: &Q, value: &V) -> &Self
     |            ^^^^^^

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `kube-client` due to 4 previous errors
warning: build failed, waiting for other jobs to finish...
```

pgopr doesn't have an explicit dependency on tracing.

What is the best way to proceed on this ? Should the version in kube be changed ?

Thanks in advance !</div>

--- a/e2e/Cargo.toml
+++ b/e2e/Cargo.toml
@@ -25,7 +25,7 @@ openssl = ["kube/openssl-tls"]
 
 [dependencies]
 anyhow = "1.0.44"
-tracing = "0.1.29"
+tracing = "0.1.36"
 tracing-subscriber = "0.3.3"
 futures = "0.3.17"
 kube = { path = "../kube", version = "^0.76.0", default-features = false, features = ["client", "runtime", "ws", "admission", "gzip"] }

--- a/examples/Cargo.toml
+++ b/examples/Cargo.toml
@@ -40,7 +40,7 @@ either = "1.6.1"
 schemars = "0.8.6"
 static_assertions = "1.1.0"
 tar = "0.4.37"
-tracing = "0.1.29"
+tracing = "0.1.36"
 tracing-subscriber = "0.3.3"
 warp = { version = "0.3", default-features = false, features = ["tls"] }
 http = "0.2.5"

--- a/kube-client/Cargo.toml
+++ b/kube-client/Cargo.toml
@@ -66,7 +66,7 @@ tame-oauth = { version = "0.7.0", features = ["gcp"], optional = true }
 pin-project = { version = "1.0.4", optional = true }
 rand = { version = "0.8.3", optional = true }
 secrecy = { version = "0.8.0", features = ["alloc", "serde"] }
-tracing = { version = "0.1.29", features = ["log"], optional = true }
+tracing = { version = "0.1.36", features = ["log"], optional = true }
 hyper-openssl = { version = "0.9.2", optional = true }
 
 [dependencies.k8s-openapi]

--- a/kube-runtime/Cargo.toml
+++ b/kube-runtime/Cargo.toml
@@ -30,7 +30,7 @@ parking_lot = "0.12.0"
 pin-project = "1.0.2"
 tokio = { version = "1.14.0", features = ["time"] }
 tokio-util = { version = "0.7.0", features = ["time"] }
-tracing = "0.1.29"
+tracing = "0.1.36"
 json-patch = "0.2.6"
 serde_json = "1.0.68"
 thiserror = "1.0.29"

